### PR TITLE
Some cleanup and refactoring

### PR DESCRIPTION
**Pull request by valery1707**

### CODE REVIEW

File: src/main/java/name/valery1707/kaitai/KaitaiException.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package name.valery1707.kaitai;
 
+@SuppressWarnings("WeakerAccess")
 public class KaitaiException extends Exception {
+	@SuppressWarnings("unused")
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiException.java]: \[Checkstyle\] WARNING: 'ctor def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiException.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
 import org.buildobjects.process.TimeoutException;
+import org.slf4j.Logger;
 
 import java.nio.file.Path;
 import java.util.LinkedHashSet;
 import java.util.Set;
 
 import static java.util.Collections.unmodifiableSet;
-import static name.valery1707.kaitai.MojoUtils.*;
+import static name.valery1707.kaitai.KaitaiUtils.*;
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Import statement for 'name\.valery1707\.kaitai\.KaitaiUtils\.\*' is in the wrong order\. Should be in the 'STATIC' group, expecting not assigned imports on this line\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Using the '\.\*' form of import should be avoided \- name\.valery1707\.kaitai\.KaitaiUtils\.\*\.

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+@SuppressWarnings("WeakerAccess")
 public class KaitaiGenerator {
 	private final Path kaitai;
 	private final Path output;
 	private final String packageName;
 	private final Set<Path> sources = new LinkedHashSet<>();
 	private boolean overwrite = false;
 
-	public static KaitaiGenerator generator(Path kaitai, Path output, String packageName) throws MojoExecutionException {
+	public static KaitaiGenerator generator(Path kaitai, Path output, String packageName) throws KaitaiException {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Line is longer than 100 characters \(found 118\)\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Missing a Javadoc comment\.

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 4]
```diff
@@ -49,13 +49,13 @@ public String getPackageName() {
 		return unmodifiableSet(sources);
 	}
 
-	public KaitaiGenerator withSource(Path source) throws MojoExecutionException {
+	public KaitaiGenerator withSource(Path source) throws KaitaiException {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Missing a Javadoc comment\.

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 5]
```diff
@@ -49,13 +49,13 @@ public String getPackageName() {
...
 		checkFileIsReadable(source);
 		this.sources.add(source);
 		return this;
 	}
 
-	public KaitaiGenerator withSource(Iterable<Path> sources) throws MojoExecutionException {
+	public KaitaiGenerator withSource(Iterable<Path> sources) throws KaitaiException {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Missing a Javadoc comment\.

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 6]
```diff
@@ -75,10 +75,8 @@ public KaitaiGenerator overwrite(boolean overwrite) {
 		return this;
 	}
 
-	private ProcBuilder process(Log log) {
+	private ProcBuilder process(Logger log) {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiGenerator.java [hunk 7]
```diff
@@ -96,7 +94,7 @@ private void execute(ProcBuilder builder) throws KaitaiException {
 		}
 	}
 
-	public Path generate(Log log) throws KaitaiException {
+	public Path generate(Logger log) throws KaitaiException {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiGenerator.java]: \[Checkstyle\] WARNING: Missing a Javadoc comment\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 8]
```diff
@@ -1,27 +1,24 @@
...
 import java.net.URL;
-import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.*;
+import java.util.Arrays;
+import java.util.List;
 
 import static java.lang.String.format;
-import static name.valery1707.kaitai.MojoUtils.*;
+import static name.valery1707.kaitai.KaitaiUtils.*;
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Import statement for 'name\.valery1707\.kaitai\.KaitaiUtils\.\*' is in the wrong order\. Should be in the 'STATIC' group, expecting not assigned imports on this line\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Import statement for 'name\.valery1707\.kaitai\.KaitaiUtils\.\*' is in the wrong order\. Should be in the 'STATIC' group, expecting not assigned imports on this line\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Using the '\.\*' form of import should be avoided \- name\.valery1707\.kaitai\.KaitaiUtils\.\*\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Using the '\.\*' form of import should be avoided \- name\.valery1707\.kaitai\.KaitaiUtils\.\*\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 9]
```diff
@@ -128,7 +123,15 @@
 	/**
 	 * Executes the plugin, to read the given source and behavioural properties and generate POJOs.
 	 */
-	public void execute() throws MojoExecutionException, MojoFailureException {
+	public void execute() throws MojoExecutionException {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 10]
```diff
@@ -128,7 +123,15 @@
...
+		try {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'try' has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 11]
```diff
@@ -128,7 +123,15 @@
...
+			executeInt();
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'try' child has incorrect indentation level 24, expected level should be 6\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 12]
```diff
@@ -128,7 +123,15 @@
...
+		} catch (KaitaiException e) {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'try rcurly' has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Catch parameter name 'e' must match pattern '^\[a\-z\]\[a\-z0\-9\]\[a\-zA\-Z0\-9\]\*$'\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 13]
```diff
@@ -128,7 +123,15 @@
...
+			throw new MojoExecutionException(e.getMessage(), e.getCause());
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'catch' child has incorrect indentation level 24, expected level should be 6\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java (hunk 11 above)
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java (hunk 12 above)
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'try rcurly' has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Catch parameter name 'e' must match pattern '^\[a\-z\]\[a\-z0\-9\]\[a\-zA\-Z0\-9\]\*$'\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java (hunk 13 above)
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'catch' child has incorrect indentation level 24, expected level should be 6\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 14]
```diff
@@ -128,7 +123,15 @@
...
+		}
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'catch rcurly' has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 15]
```diff
@@ -128,7 +123,15 @@
...
+	}
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def rcurly' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 16]
```diff
@@ -128,7 +123,15 @@
...
+
+	private void executeInt() throws KaitaiException {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 17]
```diff
@@ -154,10 +157,13 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 			return;
 		}
 
+		StaticLoggerBinder.getSingleton().setMavenLog(getLog());
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def' child has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 18]
```diff
@@ -154,10 +157,13 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+		Logger logger = StaticLoggerBinder.getSingleton().getLoggerFactory().getLogger(getClass().getName());
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def' child has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line is longer than 100 characters \(found 117\)\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 19]
```diff
@@ -154,10 +157,13 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+
 		//Download Kaitai distribution into cache and unzip it
 		URL url = prepareUrl(this.url, version);
-		Path cacheDir = prepareCache(this.cacheDir, session, getLog());
-		Path kaitai = downloadKaitai(url, cacheDir, getLog());
+		Path cacheDir = prepareCache(detectCacheDir(), logger);
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def' child has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 20]
```diff
@@ -154,10 +157,13 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+		Path kaitai = downloadKaitai(url, cacheDir, logger);
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def' child has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 21]
```diff
@@ -167,9 +173,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 				.generator(kaitai, output, packageName)
 				.withSource(source)
 				.overwrite(overwrite)
-				.generate(getLog());
+				.generate(logger);
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 22]
```diff
@@ -167,9 +173,9 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
 		} catch (KaitaiException e) {
-			throw new MojoExecutionException(
+			throw new KaitaiException(
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'catch' child has incorrect indentation level 24, expected level should be 6\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 23]
```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
-		}
-		return url;
-	}
-
-	static Path prepareCache(File target, MavenSession session, Log log) throws MojoExecutionException {
-		Path cache;
-		if (target == null) {
-			Path repository = new File(session.getLocalRepository().getBasedir()).toPath();
-			cache = repository.resolve(".cache").resolve("kaitai").normalize();
+	private Path detectCacheDir() {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'method def modifier' has incorrect indentation level 8, expected level should be 2\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 24]
```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+		if (cacheDir != null) {
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'if' has incorrect indentation level 16, expected level should be 4\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 25]
```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+			return cacheDir.toPath();
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'if' child has incorrect indentation level 24, expected level should be 6\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 26]
```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
-		download(url, distZip, log);
-		Path dist = unpack(distZip, log);
-		List<Path> bats = scanFiles(dist, new String[]{KAITAI_START_SCRIPT}, new String[0]);
-		if (bats.size() != 1) {
-			throw new MojoExecutionException(format(
-				"Fail to find start script '%s' in Kaitai distribution: %s"
-				, KAITAI_START_SCRIPT
-				, dist.normalize().toFile().getAbsolutePath()
-			));
+			Path repository = new File(session.getLocalRepository().getBasedir()).toPath();
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'else' child has incorrect indentation level 24, expected level should be 6\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line is longer than 100 characters \(found 103\)\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiMojo.java [hunk 27]
```diff
@@ -179,51 +185,12 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+			return repository.resolve(".cache").resolve("kaitai").normalize();
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: 'else' child has incorrect indentation level 24, expected level should be 6\.

Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiMojo.java]: \[Checkstyle\] WARNING: Line contains a tab character\.

File: src/main/java/name/valery1707/kaitai/KaitaiUtils.java [hunk 28]
```diff
@@ -4,41 +4,40 @@
...
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.file.*;
 import java.nio.file.attribute.BasicFileAttributes;
 import java.nio.file.attribute.PosixFilePermission;
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (sputnikci) [inline src/main/java/name/valery1707/kaitai/KaitaiUtils.java]: \[Checkstyle\] WARNING: Using the '\.\*' form of import should be avoided \- java\.util\.\*\.

**Comment by valery1707** [comment]
@rultor merge

**Comment by valery1707** [comment]
@rultor release

**Comment by valery1707** [comment]
@rultor release, tag=`0.1.2`

**Pull request merged**